site/src/main/mdoc/driver_session_transaction.md

```diff
@@ -13,9 +13,18 @@ A `Driver` is basically the connection with the Database. Usually, you would onl
 A `Session` provides a context for performing operations _(`Transactions`)_ over the database. You may need as many as concurrent operations you want to have.
 A `Transaction` is a logical container for an atomic unit of work. Only one transaction may exist in a `Session` at any point in time.
 
-## Transaction managment
+## Session thread safety
 
-Each `Transaction` has to be started, used and finally either, committed or rolled back.
+Unlike its **Java** counterpart, `neotypes.Session` is thread safe.
+We achieve that by using a simple locking mechanism that ensures only one `Transaction` can be created `Session`.
+If you want / need to run multiple queries concurrently then you need to create multiple `Sessions` and load-balance the petitions across them yourself.
```

Comment:
- I'm not sure I don't understand what `petitions` means in the current context. Did you mean something else?

- Pick one 😁 
```
`Sessions`
``` 
-> 
```
`Session`s
```